x-pack/plugins/apm/server/routes/storage_explorer/get_summary_statistics.ts

```diff
@@ -0,0 +1,223 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import {
+  termQuery,
+  kqlQuery,
+  rangeQuery,
+} from '@kbn/observability-plugin/server';
+import {
+  getTotalIndicesStats,
+  getEstimatedSizeForDocumentsInIndex,
+} from './indices_stats_helpers';
+import { Setup } from '../../lib/helpers/setup_request';
+import { ApmPluginRequestHandlerContext } from '../typings';
+import {
+  IndexLifecyclePhaseSelectOption,
+  indexLifeCyclePhaseToDataTier,
+} from '../../../common/storage_explorer_types';
+import { RandomSampler } from '../../lib/helpers/get_random_sampler';
+import {
+  SERVICE_NAME,
+  TIER,
+  INDEX,
+  TRANSACTION_ROOT,
+  PARENT_ID,
+} from '../../../common/elasticsearch_fieldnames';
+import { environmentQuery } from '../../../common/utils/environment_query';
+import {
+  getDocumentTypeFilterForTransactions,
+  getProcessorEventForTransactions,
+  getDurationFieldForTransactions,
+} from '../../lib/helpers/transactions';
+import { calculateThroughputWithRange } from '../../lib/helpers/calculate_throughput';
+
+export async function getTracesPerMinute({
+  setup,
+  indexLifecyclePhase,
+  start,
+  end,
+  environment,
+  kuery,
+  searchAggregatedTransactions,
+}: {
+  setup: Setup;
+  indexLifecyclePhase: IndexLifecyclePhaseSelectOption;
+  start: number;
+  end: number;
+  environment: string;
+  kuery: string;
+  searchAggregatedTransactions: boolean;
+}) {
+  const { apmEventClient } = setup;
+
+  const response = await apmEventClient.search('get_traces_per_minute', {
+    apm: {
+      events: [getProcessorEventForTransactions(searchAggregatedTransactions)],
+    },
+    body: {
+      size: 0,
+      track_total_hits: false,
+      query: {
+        bool: {
+          filter: [
+            ...getDocumentTypeFilterForTransactions(
+              searchAggregatedTransactions
+            ),
+            ...environmentQuery(environment),
+            ...kqlQuery(kuery),
+            ...rangeQuery(start, end),
+            ...(indexLifecyclePhase !== IndexLifecyclePhaseSelectOption.All
+              ? termQuery(
+                  TIER,
+                  indexLifeCyclePhaseToDataTier[indexLifecyclePhase]
+                )
+              : []),
+            ...(searchAggregatedTransactions
+              ? [
+                  {
+                    term: {
+                      [TRANSACTION_ROOT]: true,
+                    },
+                  },
+                ]
+              : []),
+          ] as QueryDslQueryContainer[],
+          must_not: [
+            ...(!searchAggregatedTransactions
+              ? [
+                  {
+                    exists: {
+                      field: PARENT_ID,
+                    },
+                  },
+                ]
+              : []),
+          ],
```

Comment:
Can you do something like

```suggestion
            isRootTransaction(searchAggregatedTransactions),
          ],          
```

Where 

```ts

function isRootTransaction(searchAggregatedTransactions) {
  return searchAggregatedTransactions
    ? {
        term: {
          [TRANSACTION_ROOT]: true,
        },
      }
    : {
        must_not: {
          exists: { field: PARENT_ID },
        },
      };
}
```

Reply:
@sqren Do you mean to wrap `must_not` in another `bool` query so it can be inside `filter`?

Reply:
Yes, above is pseudo-code. My intention is that we should encapsulate the logic in a helper function like above. I don't think the nested query will affect performance (I hope it gets compiled away).

Reply:
... if you do make a helper, you should update the following to use it too

https://github.com/elastic/kibana/blob/c943cedd9ea0fb0f6d79e64b8362570284387502/x-pack/plugins/apm/server/routes/traces/get_top_traces_primary_stats.ts#L83-L102

Reply:
Sure, I've changed this already. https://github.com/elastic/kibana/pull/140109/commits/a6cd1138c369d4cde2579d33615d167a2a1c575b